netweaver/monitoring.sls

```diff
@@ -6,7 +6,12 @@ prometheus_sap_host_exporter_pkg:
     - name: prometheus-sap_host_exporter
 
 # the sid, instance number pair of a node is unique, so we need to adapt configuration
-{% for node in netweaver.nodes if host == node.host and node.sap_instance != "db" %}
+# do not create configuration for "sap_instance: db"
+{% for node in netweaver.nodes if node.sap_instance != "db" %}
+# create ASCS|ERS|PAS|AAS configuration in non-HA use case
+# OR create both ASCS and ERS configuration in HA use case
+#   AND deploy PAS exporter on ASCS/PAS host if no dedicated PAS host is used
+{% if (not netweaver.ha_enabled and host == node.host) or (netweaver.ha_enabled and (node.sap_instance in ['ascs', 'ers'] or host == node.host)) %}
```

Comment:
`node.sap_instance in ['ascs', 'ers']` looks like will create the configuration file for them in all of the nodes, right? Even in the PAS and AAS nodes.
This doesn't look like really desirable (even though it is not harmful).

Reply:
Do you think we could simply using the next?:
```{% if host == node.host or (netweaver.ha_enabled and node.sap_instance in ['ascs', 'ers']) %}```

We would still have the issue of having the `ascs/ers` configuration files in the `pas/aas`, but well, who cares hehe

Reply:
> `node.sap_instance in ['ascs', 'ers']` looks like will create the configuration file for them in all of the nodes, right? Even in the PAS and AAS nodes.
> This doesn't look like really desirable (even though it is not harmful).

I could swear that this worked in my tests... But you're right, it looks suspicious now...

Reply:
> `{% if host == node.host or (netweaver.ha_enabled and node.sap_instance in ['ascs', 'ers']) %}`
> 
> We would still have the issue of having the `ascs/ers` configuration files in the `pas/aas`, but well, who cares hehe

@arbulu89 What do you think of this?
It should make sure ASCS/ERS configs are only deployed on nodes ending with `0(1|2)$` ... 
```
{% if host == node.host or (netweaver.ha_enabled and (host | regex_search('^.*0(1|2)$')) and node.sap_instance in ['ascs', 'ers']) %}
```

Reply:
I guess going for hostname matching is always a bad idea...

Reply:
Could one assume the positions in the `netweaver.nodes` array are always `["ascs","ers", "..."]`?
If this would be the case, one could use their positions.

Reply:
Hey,
one additional note here. Thinking about a diskless SBD cluster setup would mean we would have three nodes for the ASCS / ERS instances. If it would make sense to keep this feature in mind because such a deployment would be really interesting for the CSP's. Because they get rid of the SBD device as stonith method in that case. I guess this could have an impact for the solution you looking for at the moment.

Reply:
@yeoldegrove If we want to be purist and have the ASCS/ERS configuration files only on those nodes, we could even have 2 loops, the first one with 
```
{% for node in netweaver.nodes if host == node.host and node.sap_instance != "db" %}
```

And the 2nd only executed with `ascs/ers` to find if there is any other `ascs/ers` to ship the config file to them too.
It looks costly, but maybe we need to do it

Reply:
@arbulu89 fixed in 4ffafae

Reply:
@arbulu89 the third node which I had in mind is a valid SAP node which could run the ASCS or ERS. Such a scenario has many advantages for customer and the CSP setups. This node is not limited to be only a quorum node.

Reply:
Ok, so if this is the case I'm afraid that the monitoring is the last thing that worries me, because most probably the rest of the deployment would fail and we would need to do major changes

Reply:
You are right and I agree. 
If we did not plan to support such a scenario we do not need to take care for this circumstance for the monitoring feature. My point was just keep in mind that we have a supported solution which is interesting for Cloud setups (the request came initially from Azure) and is based on three or five or more nodes. Initially the ASCS and ERS are only installed / available on two nodes but could run on all the other nodes as well.